lambdas/postgres-migration-async-operation/iam.tf

```diff
@@ -0,0 +1,109 @@
+data "aws_iam_policy_document" "migration_async_operation_assume_role_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_security_group" "postgres_migration_async_operation" {
+  count = length(var.lambda_subnet_ids) == 0 ? 0 : 1
+
+  name   = "${var.prefix}-postgres-migration-async-operation"
+  vpc_id = var.vpc_id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = var.tags
+}
+
+# postgres_migration_async_operation_role
+resource "aws_iam_role" "postgres_migration_async_operation_role" {
+  name                 = "${var.prefix}-postgres_migration_async_operation"
+  assume_role_policy   = data.aws_iam_policy_document.migration_async_operation_assume_role_policy.json
+  permissions_boundary = var.permissions_boundary_arn
+  tags                 = var.tags
+}
+
+data "aws_iam_policy_document" "postgres_migration_async_operation_policy" {
+  statement {
+    actions = [
+      "ecs:RunTask",
+      "ec2:CreateNetworkInterface",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "dynamodb:ListTables",
+      "lambda:GetFunction",
+      "lambda:GetFunctionConfiguration",
```

Comment:
Interesting (this may be answered later in the review) - what's this used for?

Reply:
https://github.com/nasa/cumulus/blob/feature-rds/packages/async-operations/src/async_operations.ts#L21

It's eventually used to start the ECS task! 
```
startAsyncOperation -> startECSTask -> getLambdaEnvironmentVariables
```

Reply:
Though, taking a closer look, I may be putting that permission on the wrong policy document 🤔

Reply:
If the lambda is what's invoking that, I think you are.   I just noted the difference from other similar lambda configurations, and that explains the additional perms.  Thanks!